### PR TITLE
Remove remaining Overlapped usage from System.IO.Pipes

### PR DESCRIPTION
**Pull request by stephentoub**
Related to #1972

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @ericstj, @weshaggard, @davkean, @ericeil

File: src/System.IO.Pipes/src/System/IO/Pipes/PipeAsyncResult.cs [hunk 1]
```diff
@@ -9,12 +9,11 @@ namespace System.IO.Pipes
 {
     internal unsafe sealed class PipeAsyncResult : IAsyncResult
     {
+        internal NamedPipeServerStream _stream;
```
Comment (ericstj) [inline src/System.IO.Pipes/src/System/IO/Pipes/PipeAsyncResult.cs]: Why not just make this a ThreadPoolBoundHandle?  Do you need to root the NamedPipeServerStream?

Comment (stephentoub) [inline src/System.IO.Pipes/src/System/IO/Pipes/PipeAsyncResult.cs]: Sure, that should work, too.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@ericstj, any other feedback?

**Comment by ericstj** [comment]
LGTM :+1:

**Comment by stephentoub** [comment]
Excellent, thanks.

**Pull request merged**